### PR TITLE
[W5.7][T09-3]Benedict

### PR DESCRIPTION
**Pull request by benedictcss**

### CODE REVIEW

File: src/seedu/addressbook/data/AddressBook.java [hunk 1]
```diff
@@ -54,6 +54,13 @@ public void removePerson(ReadOnlyPerson toRemove) throws PersonNotFoundException
         allPersons.remove(toRemove);
     }
 
+    /**
+     * Clears all persons and tags from the address book.
```
Comment (ihwk1996) [inline src/seedu/addressbook/data/AddressBook.java]: This comment should be *Sort all persons from the address book

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (ihwk1996) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Isnt it bad to import *??

**Review by ihwk1996 — COMMENTED** [review]
Overall good :)

**Comment by lancelotzty** [comment]
it is quite clean, so good

**Comment by j-lum** [comment]
Everything looks good to me. Might want to put a PR description next time.

**Comment by chivent** [comment]
nice and clean! test cases are well updated as well

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Sorts the address book alphabetically.
+ */
+public class SortCommand extends Command {
+    public static final String COMMAND_WORD = "sort";
+    public static final String MESSAGE_USAGE = "Sorts the address book alphabetically.\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Address book has been sorted!";
```
Comment (eugenepeh) [inline src/seedu/addressbook/commands/SortCommand.java:11]: Good, but if this constant is only going to be used internally, perhaps can use private accessor instead

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 4]
```diff
@@ -122,6 +123,18 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
         }
     }
 
+    /**
+     * Sorts all persons in list.
+     */
+    public void sort() {
+        internalList.sort(new Comparator<Person>() {
```
Comment (eugenepeh) [inline src/seedu/addressbook/data/person/UniquePersonList.java:130]: Good! Another way would be to shorten it using a lambda expression.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 5]
```diff
@@ -10,6 +10,25 @@
 
 public class UtilsTest {
 
+    @Test
+    public void isAnyNull() {
```
Comment (eugenepeh) [inline test/java/seedu/addressbook/common/UtilsTest.java:14]: This shouldn't be part of this PR. Please create a clean PR everytime you try to add/make a new change.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import static org.junit.Assert.assertNotEquals;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.*;
```
Comment (eugenepeh) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:13]: remember to disable intellij auto import wildcard
[Why is wild card import bad?](https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad)

**Review by eugenepeh — APPROVED** [review]
Good job on completing the activity.

You've updated all the aspects, docs, tests etc...
However try not to include features/changes that not supposed to be part of the PR.

Some comments added. Please close the PR after reading the comments.
